### PR TITLE
Add sss for hypershift tenant config

### PR DESCRIPTION
**Pull request by wanghaoran1988**
### What type of PR is this?
feature
### What this PR does / why we need it?
We need use a sss to sync the hypershift credential to all management clusters.
### Which Jira/Github issue(s) this PR fixes?

_Fixes #_ [OSD-13723](https://issues.redhat.com//browse/OSD-13723)

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [x] Included documentation changes with PR
- [x] If this is a new object that is not intended for the FedRAMP environment (if unsure, please reach out to team FedRAMP), please exclude it with:

    ```yaml
    matchExpressions:
    - key: api.openshift.com/fedramp
      operator: NotIn
      values: ["true"]
    ```

### CODE REVIEW

File: hack/00-osd-managed-cluster-config-integration.yaml.tmpl [hunk 1]
```diff
@@ -18466,3 +18466,28 @@ objects:
...
+        namespace: hive
+      targetRef:
+        name: rhobs-hypershift-credential
+        namespace: openshift-monitoring
+    - sourceRef:
+        name: rhobs-hypershift-credential
+        namespace: hive
+      targetRef:
+        name: rhobs-hypershift-credential
+        namespace: openshift-observability-operator
```
Comment (patjlm) [inline hack/00-osd-managed-cluster-config-integration.yaml.tmpl:18493]: Are you sure you want to use the same secret in both locations ?

Comment (patjlm) [inline hack/00-osd-managed-cluster-config-integration.yaml.tmpl:18493]: How will the namespace(s) be created? Is it during the operator deployment ?

Comment (wanghaoran1988) [reply hack/00-osd-managed-cluster-config-integration.yaml.tmpl:18493]: yes, the one in openshift-monitoring will be used to confiure a remoteWrite to ship management cluster metrics.

File: deploy/rhobs-hypershift-config/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# RHOBS hypershift config
```
Comment (supreeth7) [inline deploy/rhobs-hypershift-config/README.md:1]:
```suggestion
# RHOBS HyperShift configuration
```

File: deploy/rhobs-hypershift-config/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+
+This dirrectory contains a syncset that copy the hypershift tenant configs into all management clusters.
```
Comment (supreeth7) [inline deploy/rhobs-hypershift-config/README.md:3]:
```suggestion
This directory contains a SyncSet that copies the HyperShift RHOBS tenant configuration into all management clusters.
```

**Comment by mrbarge** [comment]
Can you add an `OWNERS` file to the added directory which defines who should be the reviewers of changes? It is an established convention in most other `deploy` subdirectories.

**Review by supreeth7 — CHANGES_REQUESTED** [review]
Found a few minor nits, none breaking though

**Pull request merged**